compiler_opt/es/blackbox_learner.py

```diff
@@ -225,11 +235,31 @@ def _log_tf_summary(self, rewards: list[float]) -> None:
           len(train_wins) / len(rewards),
           step=self._step)
 
-  def _save_model(self) -> None:
+  def _save_model(self, parameters: npt.NDArray[np.float32],
+                  policy_name: str) -> None:
     """Save the model."""
     logging.info('Saving the model.')
-    self._policy_saver_fn(
-        parameters=self._model_weights, policy_name=f'iteration{self._step}')
+    self._models_to_save.append((parameters, policy_name))
+
+  def _start_model_saving(self):
+    for _ in range(len(self._models_to_save)):
+      model_parameters, model_name = self._models_to_save[0]
+      self._models_to_flush.append(
+          self._thread_pool.apply_async(self._policy_saver_fn,
+                                        (model_parameters, model_name)))
+      self._models_to_save.pop(0)
+
+  def _flush_models(self):
+    for _ in range(len(self._models_to_flush)):
+      model_save_result = self._models_to_flush[0]
+      model_save_result.wait()
+      if not model_save_result.successful():
```

Comment:
*not* successful?

Reply:
oh you want to raise the exception? doesn't a future have an exception you could raise?